### ISSUE TITLE
ForecastDFDataset setting autoregressive_modeling False leads to no variety in model predictions

### ISSUE THREAD

**Issue by JamieToddUK** [ibm-granite/granite-tsfm#47]
My time series dataset consists of many input columns and a few target columns. I aim to train a PatchTSMixer model and PatchTST model to forecast the target columns given the previous values of the target columns in the context window, and then to train a second PatchTSMixer and PatchTST model to forecast the same target columns, but when not given the previous target values in the context.

I noticed that ForecastDFDataset has a boolean argument called "autoregressive_modeling", which when set to False, masks the targets values in the context window by setting them to 0. I believe this is what I'm looking for regarding my latter task. If I'm mistaken, please let me know.

When training my models with `autoregressive_modeling = True`, there are no issues and the models perform well. However, when setting `autoregressive_modeling = False` for the same dataset and training the models, my models only predict 0 on the test set for all target columns and time steps with minimal variance.

I've tried possible solutions such as changing the mask value to 1, but then the models predict only 1 for all target columns. I've also tried leaving `autoregressive_modeling = True` for the training set, and False for the validation/test set, however this lead to the same issue.

The target columns are not part of the "conditional columns" and so future target values are not masked.

For context I am using a TimeSeriesPreprocessor with `scaling = True`. I am training for 1 epoch using a learning rate of 0.001 (though I tried 5 epochs to no success). My PatchTSMixer and PatchTST have the following configs:

```
PatchTSMixerConfig(
    context_length = 128,
    prediction_length = 64,
    patch_length = 8,
    patch_stride = 8,
    num_input_channels = 20,
    d_model = 16,
    num_layers = 8,
    expansion_factor = 2,
    dropout = 0.1,
    head_dropout = 0.1,
    mode = "common_channel",
    scaling = "std"
)
```
```
PatchTSTConfig(
    context_length = 128,
    prediction_length = 64,
    patch_length = 8,
    patch_stride = 8,
    num_input_channels = 24,
    random_mask_ratio = 0.4,
    d_model = 128,
    ffn_dim = 256,
    num_attention_heads = 16,
    num_hidden_layers = 3,
    dropout = 0.2,
    head_dropout = 0.2,
    scaling = "std",
    pooling_type = None,
    channel_attention = False,
    pre_norm = True,
    norm_type = "batchnorm"
)
```

For inference on the test set, I extract predictions by doing the following:
```
predictions = trainer.predict(dataset)
pred_values = predictions.predictions[0][..., :num_targets]
true_values = predictions.label_ids[..., :num_targets]
```

Any help will be greatly appreciated. If the `autoregressive_modeling` boolean has a different purpose than what I thought, then let me know and please provide some guidance of how I could use these models to forecast time series without given the previous target values in the context window. Thanks in advance.

**Comment by wgifford** [comment]
@JamieToddUK 
The `autoregressive_modeling` option is experimental -- we quickly added it to see if we could limit the autoregressive tendencies of the model, for example, in cases where one wants to learn relationships between the target and other variables, rather the past target values.

We had similar conclusions as above -- generally, `autoregressive_modeling=False` did not work well. It is likely to be removed in the future.

**Comment by JamieToddUK** [comment]
@wgifford No worries, I understand. Thanks for the help.
